osu.Game/Screens/Play/SpectatorPlayer.cs

```diff
@@ -3,33 +3,61 @@
 
 using System.Linq;
 using osu.Framework.Allocation;
+using osu.Framework.Graphics;
 using osu.Framework.Screens;
 using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
 using osu.Game.Online.Spectator;
 using osu.Game.Scoring;
+using osu.Game.Screens.Ranking;
 
 namespace osu.Game.Screens.Play
 {
-    public class SpectatorPlayer : ReplayPlayer
+    public class SpectatorPlayer : Player
     {
-        [Resolved]
-        private SpectatorStreamingClient spectatorStreaming { get; set; }
+        private readonly Score score;
+
+        protected override bool CheckModsAllowFailure() => false; // todo: better support starting mid-way through beatmap
 
         public SpectatorPlayer(Score score)
-            : base(score)
+            : base(true, true)
```

Comment:
Rider seems to tell me that this `base` delegation is redundant (and so it appears to be).